argo-data-submission-wf/main.nf

```diff
@@ -166,38 +189,103 @@ process checkCramReference{
+    } else if (
+      metadata_payload_json.startsWith("NO_FILE") && \
+      data_directory.startsWith("NO_FILE") && \
+      (og_experiment_info_tsv.startsWith("NO_FILE") || og_read_group_info_tsv.startsWith("NO_FILE") || og_file_info_tsv.startsWith("NO_FILE"))
+    ){
+      exit 1,"`experiment_info_tsv` , `read_group_info_tsv` , and `file_info_tsv` must be invoked together."
+    } else if (
+      !og_experiment_info_tsv.startsWith("NO_FILE") && \
+      !og_read_group_info_tsv.startsWith("NO_FILE") && \
+      !og_file_info_tsv.startsWith("NO_FILE") && \
+      !metadata_payload_json.startsWith("NO_FILE") && \
+      !data_directory.startsWith("NO_FILE")
+    ) {
+      exit 1,"Too many parameters invoked. Please re-submit with either of the following pairings: 1. `experiment_info_tsv` , `read_group_info_tsv` , and `file_info_tsv` 2. `metadata_payload_json` and `data_directory`"
+    }
+
+    if (
+      og_experiment_info_tsv.startsWith("NO_FILE") && \
+      og_read_group_info_tsv.startsWith("NO_FILE") && \
+      og_file_info_tsv.startsWith("NO_FILE")
+    ) {
+      payloadJsonToTsvs(
+        file(metadata_payload_json),
+        file(data_directory)
+        )
+
+      sanityCheck(
+        payloadJsonToTsvs.out.experiment_tsv,
+        api_token,
+        song_url,
+        clinical_url,
+        params.skip_duplicate_check
+      )
+      
+      experiment_info_tsv=sanityCheck.out.updated_experiment_info_tsv
+      read_group_info_tsv=payloadJsonToTsvs.out.read_group_tsv
+      file_info_tsv=payloadJsonToTsvs.out.file_tsv
+    } else {
+      sanityCheck(
+        file(og_experiment_info_tsv),
+        api_token,
+        song_url,
+        clinical_url,
+        params.skip_duplicate_check
+      )
+      
+      experiment_info_tsv=sanityCheck.out.updated_experiment_info_tsv
+      read_group_info_tsv=file(og_read_group_info_tsv)
+      file_info_tsv=file(og_file_info_tsv)
     }
 
     checkCramReference(
-      file(file_info_tsv),
-      file(ref_genome_fa)
+      file_info_tsv,
+      file(ref_genome_fa),
+      experiment_info_tsv
       )
 
 
     // download from ega after payload is generated and valid according to the given schema
-    if (params.download_mode!='local'){
+    if (params.download_mode!='local' ){
       egaWf(
         params.download_mode,
         file_info_tsv,
         checkCramReference.out.check_status
       )
       sequence_files=egaWf.out.sequence_files
+    } else if (og_file_info_tsv.startsWith("NO_FILE")){
+      sequence_files=file_info_tsv.splitCsv( header : true , sep:'\t').map( row -> file("${row.path}",checkIfExists : true)) 
     } else {
-      sequence_files=Channel.fromPath(file_info_tsv) | splitCsv( header : true , sep:'\t') | map( row -> file("${row.path}",checkIfExists : true)) 
+      sequence_files=Channel.fromPath(file_info_tsv).splitCsv( header : true , sep:'\t').map( row -> file("${row.path}",checkIfExists : true))
```

Comment:
Just curious, what's the reason for differential `og_file_info_tsv.startsWith("NO_FILE")`?

Reply:
Naming scheme was to track original `experiment.tsv` and updated version. The usage in `file_info` and `read_groups` was to stay consistent.